utils/js/LangUtils.ts

```diff
@@ -283,6 +284,15 @@ export function filterConsecutive<T>(
     };
 }
 
+/**
+ * Intersperse a separator between each item in an array.
+ */
+export function intersperse<T>(arr: T[], separator: any): T[] {
```

Comment:
nifty!  I have always wanted such a thing.  Not sure about the type.  Seems like Seperator should be also of type `T`.  Or alternatively,  could have another parameter `S` and have it return `Array<T|S>`